src/Features/Core/Portable/Completion/Providers/AbstractInternalsVisibleToCompletionProvider.cs

```diff
@@ -0,0 +1,218 @@
+                var attributeSyntaxNodeCandidate = node.Parent?.Parent?.Parent;
+                if (syntaxFactsService.IsAttribute(attributeSyntaxNodeCandidate))
+                {
+                    return attributeSyntaxNodeCandidate;
+                }
+            }
+
+            return null;
+        }
+
+        private static async Task<bool> CheckTypeInfoOfAttributeAsync(Document document, SyntaxNode attributeNode, CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetSemanticModelForNodeAsync(attributeNode, cancellationToken).ConfigureAwait(false);
+            var typeInfo = semanticModel.GetTypeInfo(attributeNode);
+            var type = typeInfo.Type;
+            if (type == null)
+            {
+                return false;
+            }
+
+            var internalsVisibleToAttributeSymbol = semanticModel.Compilation.GetTypeByMetadataName(typeof(InternalsVisibleToAttribute).FullName);
+            return type.Equals(internalsVisibleToAttributeSymbol);
+        }
+
+        private async Task AddAssemblyCompletionItemsAsync(CompletionContext context, CancellationToken cancellationToken)
+        {
+            var currentProject = context.Document.Project;
+            var allInternalsVisibleToAttributesOfProject = await GetAllInternalsVisibleToAssemblyNamesOfProjectAsync(context, cancellationToken).ConfigureAwait(false);
+            var currentCompilation = await currentProject.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+            var currentAssemblyInfo = currentCompilation.Assembly;
+            foreach (var project in context.Document.Project.Solution.Projects)
+            {
+                if (project == currentProject)
+                {
+                    continue;
+                }
+
+                if (allInternalsVisibleToAttributesOfProject.Contains(project.AssemblyName) == true)
+                {
+                    continue;
+                }
+
+                var projectGuid = project.Id.Id.ToString();
+                var completionItem = CommonCompletionItem.Create(
+                    displayText: project.AssemblyName,
+                    rules: CompletionItemRules.Default,
+                    glyph: project.GetGlyph(),
+                    properties: ImmutableDictionary.Create<string, string>().Add(ProjectGuidKey, projectGuid));
+                context.AddItem(completionItem);
+            }
+        }
+
+        private async Task<IImmutableSet<string>> GetAllInternalsVisibleToAssemblyNamesOfProjectAsync(CompletionContext completionContext, CancellationToken cancellationToken)
+        {
+            var project = completionContext.Document.Project;
+            var resultBuilder = default(ImmutableHashSet<string>.Builder);
+            foreach (var document in project.Documents)
+            {
+                var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
I did some (unscientific) measuring with 
```
var sw = new Stopwatch();
sw.Start();
var allInternalsVisibleToAttributesOfProject = await GetAllInternalsVisibleToAssemblyNamesOfProjectAsync(context, cancellationToken).ConfigureAwait(false);
File.AppendAllText(@"E:\Measure.txt", "allInternalsVisibleToAttributesOfProject: " + sw.Elapsed.ToString("G") + "\n");
```
So this was measured with a debug build of VS and without a debugger attached (Ctrl+F5). Performance of this code path is depended on the number of documents in the current project and not the number of projects in the solution. Therefore I looked for projects with a huge number of documents. 
I measured in  2 solutions and three class libraries:
1. Solution (Own project)
   1. Project with 35 cs documents
   2. Project with 285 cs documents
2. Roslyn.sln
   1. Workspaces.csproj with 900 cs documents

(Number of documents = count of `<Compile Include="" />` in csproj)

I did 5 runs with the following results (seconds):

| 35 documents | 285 documents | 900 documents (Workspaces.csproj) |
|--------------|---------------|-----------------------------------|
| 0,0036259    | 0,0210543     | 0,0298217                         |
| 0,003596     | 0,0126806     | 0,0216452                         |
| 0,0035024    | 0,0046144     | 0,0327513                         |
| 0,0036021    | 0,0168759     | 0,0211471                         |
| 0,0037565    | 0,0048232     | 0,0271816                         |


As diagram:
![diagram](https://user-images.githubusercontent.com/24472128/28115478-6f385170-6705-11e7-88f3-406ec817525b.png)

The maximum duration measured was 33ms.

This is how ist looks like in action:
![performance](https://user-images.githubusercontent.com/24472128/28115457-5c26ef38-6705-11e7-8297-1f377f04a4ae.gif)

IMHO this is fast enough. I haven't measured other indicators (number and size of heap allocations and the like) nor visual basic.